### PR TITLE
HLTElectronMuonInvMassFilter update

### PR DESCRIPTION
**Pull request by gpasztor**
The HLTElectronMuonInvariantMassFilter code is modified to accept also RecoEcalCandidate - Muon pairs in addition to Electron - Muon. Modification made and tested by Matt Carver (following the suggestion of @matteosan1 ).

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
Since we now get more possible objects, the accept rate will go up.
Has this been studied on signal and esp. background (HLT rate)?

**Comment by matteosan1** [comment]
Hi Martin,
as far as I know there will never be a a mixed situation where you can can
pair to muons, electrons or photons.
The change has been done to let few paths to cope with the new EGM
convention which propagate down the path just RecoEcalCandidate: see for
example HLT_Mu8_Ele8_CaloIdM_GsfTrackIdM_Mass8_PFHT300


Regards,
     Matteo.

+++++++++++
Matteo Sani
University of California, San Diego, US.
Tel: +41227671577
CERN Office: 40-3-A02

**Comment by Martin-Grunewald** [comment]
+1
OK, thanks for the clarification.

File: HLTrigger/Egamma/src/HLTElectronMuonInvMassFilter.cc [hunk 1]
```diff
@@ -102,8 +115,18 @@ HLTElectronMuonInvMassFilter::hltFilter(edm::Event& iEvent, const edm::EventSetu
     pElectron.push_back( pThisEle );
     eleCharge.push_back( refele->charge() );
   }
+  
+  for(unsigned int i=0;i<clusCands.size();i++){
```
Comment (davidlange6) [inline HLTrigger/Egamma/src/HLTElectronMuonInvMassFilter.cc]: please use reserve on pElectron and eleCharge  to pre-allocate the memory here.
(eg, pElectron.reserve(clusCands.size()); )

File: HLTrigger/Egamma/src/HLTElectronMuonInvMassFilter.cc [hunk 2]
```diff
@@ -117,6 +140,20 @@ HLTElectronMuonInvMassFilter::hltFilter(edm::Event& iEvent, const edm::EventSetu
       }
     }
   }
+  
+  for(unsigned int i=0; i<clusCands.size(); i++) {
+    for(unsigned int j=0; j<l3muons.size(); j++) {
+      TLorentzVector p1 = pElectron.at(i);
+      TLorentzVector p2 = pMuon.at(j);
+      TLorentzVector pTot = p1 + p2;
```
Comment (davidlange6) [inline HLTrigger/Egamma/src/HLTElectronMuonInvMassFilter.cc]: please replace by TLorentzVector pTot= pElectron[i] + pMuon[j];
@gpasztor

**Comment by Martin-Grunewald** [comment]
please test

**Comment by fwyzard** [comment]
A rather general comment: why do you use `TLorentzVector` instead of `math::XYZTLorentzVector` ?
You can get the latter for free from the Candidates' `p4()` method, without the need to create intermediate objects.

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by Martin-Grunewald** [comment]
@davidlange6 
What is holding up this PR?

**Comment by Martin-Grunewald** [comment]
please test

**Comment by davidlange6** [comment]
+1

**Pull request merged**